azurerm/internal/services/eventhub/eventhub_resource.go

```diff
@@ -144,6 +144,23 @@ func resourceEventHub() *pluginsdk.Resource {
 				},
 			},
 
+			"status": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(eventhubs.EntityStatusActive),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(eventhubs.EntityStatusActive),
+					string(eventhubs.EntityStatusCreating),
+					string(eventhubs.EntityStatusDeleting),
+					string(eventhubs.EntityStatusDisabled),
+					string(eventhubs.EntityStatusReceiveDisabled),
+					string(eventhubs.EntityStatusRenaming),
+					string(eventhubs.EntityStatusRestoring),
+					string(eventhubs.EntityStatusSendDisabled),
+					string(eventhubs.EntityStatusUnknown),
+				}, false),
+			},
```

Comment:
what happened when you set this property to creating or deleting? this doesn't seem likea useful property that makes sense to set? more something the service exposes?

Reply:
After checked, Azure portal and [azure cli](https://docs.microsoft.com/zh-cn/cli/azure/eventhubs/eventhub?view=azure-cli-latest) only exposed the `Active`, `Disabled` and `SendDisabled`. I assume they are more useful to user. So I've updated code to only support them.
![image](https://user-images.githubusercontent.com/19754191/120727787-fd4f2000-c50d-11eb-80bc-013ac760e717.png).

Reply:
@katbyte , after confirmed with service team, `Active`, `Disabled` and `SendDisabled` are the only states that the service currently uses for Microsoft.EventHub/namespaces/eventHubs resource. So we just need to support these three values.